Development/Configuration.md

```diff
@@ -142,42 +142,66 @@ An example is shown below
+OIDC_CLIENT_SECRET=my-super-scicat-secret-123 // Client Secret to go with the above id
+OIDC_CALLBACK_URL=https://scicat.myorg.com/api/v3/auth/oidc/callback // callback URL for OIDC
+OIDC_SCOPE='openid profile email' // Scopes as required by Scicat
+OIDC_SUCCESS_URL=https://scicat.myorg.com/auth-callback // The URL to redirect to after a successful login
+OIDC_FAILURE_URL=https://scicat.myorg.com/login // *Optional* The URL to redirect to after a unsuccessful login
+OIDC_ACCESS_GROUPS=accessgroups // *Optional* scopes to retrieve access groups from OIDC
+
+//*Optional* - for adding a logbook to Scicat
 LOGBOOK_ENABLED=<"yes"|"no">  // *Optional* Flag to enable/disable the Logbook endpoints. Values "yes" or "no". Defaults to "no".
 LOGBOOK_BASE_URL="http://localhost:3030/scichatapi"  // *Optional* The base URL to the SciChat wrapper API. Only required if Logbook is enabled.
 LOGBOOK_USERNAME="<LOGBOOK_USERNAME>"  // *Optional* The username used to authenticate to the SciChat wrapper API. Only required if Logbook is enabled.
 LOGBOOK_PASSWORD="<LOGBOOK_PASSWORD>"  // *Optional* The password used to authenticate to the SciChat wrapper API. Only required if Logbook is enabled.
+
+//*Optional* - for data
 METADATA_KEYS_RETURN_LIMIT=100  // *Optional* The return limit for the `/Datasets/metadataKeys` endpoint.
 METADATA_PARENT_INSTANCES_RETURN_LIMIT=100  // *Optional* The return limit of Datasets to extract metadata keys from for the `/Datasets/metadataKeys` endpoint.
-MONGODB_URI="mongodb://<USERNAME>:<PASSWORD>@<HOST>:27017/<DB_NAME>"  // The URI for your MongoDB instance.
 OAI_PROVIDER_ROUTE="<OAI_PROVIDER_ROUTE>"  // *Optional* URI to OAI provider, used for the `/publisheddata/:id/resync` endpoint.
-PID_PREFIX="<PID_PREFIX>"  // The facility PID prefix, with trailing slash.
-PUBLIC_URL_PREFIX="https://doi.ess.eu/detail/"  // The base URL to the facility Landing Page.
-PORT=3000  // *Optional* The port on which you want to access the app. Defaults to `3000`.
+REGISTER_DOI_URI="https://mds.test.datacite.org/doi"  // URI to the organization that registers the facilities DOIs.
+REGISTER_METADATA_URI="https://mds.test.datacite.org/metadata"  // URI to the organization that registers the facilities published data metadata.
+
+//*Optional* - for connecting to AMQP service RabbitMQ
 RABBITMQ_ENABLED=<"yes"|"no">  *Optional* Flag to enable/disable RabbitMQ consumer. Values "yes" or "no". Defaults to "no".
 RABBITMQ_HOSTNAME="localhost"  // *Optional* The hostname of the RabbitMQ message broker. Only required if RabbitMQ is enabled.
 RABBITMQ_USERNAME="rabbitmq"  // *Optional* The username used to authenticate to the RabbitMQ message broker. Only required if RabbitMQ is enabled.
 RABBITMQ_PASSWORD="rabbitmq"  // *Optional* The password used to authenticate to the RabbitMQ message broker. Only required if RabbitMQ is enabled.
-REGISTER_DOI_URI="https://mds.test.datacite.org/doi"  // URI to the organization that registers the facilities DOIs.
```

Comment:
Does this really go away?

Reply:
it has been moved to line 187. In the old config almost everything was labelled as optional , even though if you want to use an optinal feature, say OIDC, the OIDC_CLIENT_ID is not optional.  I have restructured config to try and put the features in  optional blocks and indciate in those blocks when a setting is optional/required for that feature. The config logic I was going for:

SCICAT_REQUIRED_VAR_A
SCICAT_REQUIRED_VAR_B

//*Optional*- Fature to use in scicat
VARIABLE
OPTIONAL_VARIABLE //*optional*


Happy to change back.